homeassistant/components/switch/pilight.py

```diff
@@ -116,26 +146,38 @@ def _handle_code(self, call):
         # - Call turn on/off only once, even if more than one code is received
         if any(self._code_on_receive):
             for on_code in self._code_on_receive:
-                if on_code.items() <= call.data.items():
-                    self.turn_on()
+                if on_code.match(call.data):
+                    on_code.run(switch=self, turn_on=True)
                     break
 
         if any(self._code_off_receive):
             for off_code in self._code_off_receive:
-                if off_code.items() <= call.data.items():
-                    self.turn_off()
+                if off_code.match(call.data):
+                    off_code.run(switch=self, turn_on=False)
                     break
 
+    def set_state(self, turn_on, send_code=True):
```

Comment:
This code is confusing. For your feature you need 2 additional `if-conditions`, not an extra function and an extra class.